src/v/cluster/data_migration_backend.cc

```diff
@@ -1434,7 +1451,25 @@ backend::get_work_scope(const migration_metadata& metadata) {
     return std::visit(
       [&metadata](const auto& migration) {
           migration_direction_tag<std::decay_t<decltype(migration)>> tag;
-          return get_work_scope(tag, metadata);
+          auto scope = get_work_scope(tag, metadata);
+          if (migration.auto_advance && !scope.sought_state) {
+              switch (metadata.state) {
+              case state::planned:
+                  scope.sought_state = state::preparing;
+                  break;
+              case state::prepared:
+                  scope.sought_state = state::executing;
+                  break;
+              case state::executed:
+                  scope.sought_state = state::cut_over;
+                  break;
+              case state::finished:
+                  scope.sought_state = state::deleted;
+                  break;
+              default:;
```

Comment:
maybe use break; or [[fallthrough]]; ? i'd guess this would compile, but visually it looks like a typo.